cclib/method/ddec.py

```diff
@@ -300,6 +327,324 @@ def calculate_refcharges(self):
 
             # In DDEC6, weights of 1/3 and 2/3 are assigned for stockholder and localized charges.
             # (Equation 50 and 58 in doi: 10.1039/c6ra04656h)
-            refcharges[atomi] = (stockholdercharges[atomi] / 3.0) + (localizedcharges[atomi] * 2.0 / 3.0)
-        
+            refcharges[atomi] = (stockholdercharges[atomi] / 3.0) + (
+                localizedcharges[atomi] * 2.0 / 3.0
+            )
+
         return refcharges, localizedcharges, stockholdercharges
+
+    def condition_densities(self):
+        # Generator object to iterate over the grid
+        xshape, yshape, zshape = self.chgdensity.data.shape
+        atoms = len(self.data.atomnos)
+        indices = (
+            (i, x, y, z)
+            for i in range(atoms)
+            for x in range(xshape)
+            for y in range(yshape)
+            for z in range(zshape)
+        )
+
+        self._rho_ref = numpy.zeros((xshape, yshape, zshape))
+
+        for atomi, xindex, yindex, zindex in indices:
+            # rho_ref -- Equation 41 in doi: 10.1039/c6ra04656h
+            self._rho_ref[xindex][yindex][zindex] = (
+                self._rho_ref[xindex][yindex][zindex]
+                + self.proatom_density[atomi][self.closest_r_index[atomi][xindex][yindex][zindex]]
```

Comment:
Can you do a += here?

Reply:
Yes! I've tried to change any other similar cases as well. Thanks!